src/Microsoft.DotNet.CodeFormatting/FormattingEngineImplementation.cs

```diff
@@ -196,7 +205,43 @@ public ImmutableArray<DiagnosticDescriptor> AllSupportedDiagnostics
             var watch = new Stopwatch();
             watch.Start();
 
-            var diagnostics = await GetDiagnostics(project, cancellationToken).ConfigureAwait(false);
+            var workspace = project.Solution.Workspace;
+
+            await FormatProjectWithSyntaxAnalyzersAsync(workspace, project.Id, cancellationToken);
+            await FormatProjectWithLocalAnalyzersAsync(workspace, project.Id, cancellationToken);
+            await FormatProjectWithGlobalAnalyzersAsync(workspace, project.Id, cancellationToken);
+
+            watch.Stop();
+            FormatLogger.WriteLine("Total time for formatting {0} - {1}", project.Name, watch.Elapsed);
+        }
+
+        private async Task FormatProjectWithSyntaxAnalyzersAsync(Workspace workspace, ProjectId projectId, CancellationToken cancellationToken)
+        {
+            var analyzers = _analyzers.Where(a => a.SupportedDiagnostics.All(d => d.CustomTags.Contains(RuleType.Syntactic)));
```

Comment:
Yes I had forgotten about this aspect. When we add the switch we could choose to be either conservative (if an analyzer doesn't have one of these tags then just treat is as a GlobalSemantic analyzer and apply them one by one) or be very lax (treat them as LocalSemantic analyzers and apply them together and they could break the code).